samples/api/texture_mipmap_generation/texture_mipmap_generation_tutorial.md

```diff
@@ -0,0 +1,224 @@
+
+***Note:*** Blitting (same for copying) images is done inside of a command buffer that has to be submitted and as such has to be synchronized before using the new image with e.g. a ```vkFence```. 
+
+We simply loop over all remaining mip levels (level 0 was loaded from disk) and prepare a ```VkImageBlit``` structure for each blit from mip level i-1 to level i.
+
+First the source for our blit. This is the previous mip level. The dimensions of the blit source are specified by srcOffset:
+```cpp
+for (int32_t i = 1; i < texture.mipLevels; i++)
+{
+  VkImageBlit image_blit{};
+
+  // Source
+  image_blit.srcSubresource.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
+  image_blit.srcSubresource.layerCount = 1;
+  image_blit.srcSubresource.mipLevel   = i - 1;
+  image_blit.srcOffsets[1].x           = int32_t(texture.width >> (i - 1));
+  image_blit.srcOffsets[1].y           = int32_t(texture.height >> (i - 1));
+  image_blit.srcOffsets[1].z           = 1;
+```
+Setup for the destination mip level (1), with the dimensions for the blit destination specified in dstOffsets[1]:
+```cpp
+// Destination
+image_blit.dstSubresource.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
+image_blit.dstSubresource.layerCount = 1;
+image_blit.dstSubresource.mipLevel   = i;
+image_blit.dstOffsets[1].x           = int32_t(texture.width >> i);
+image_blit.dstOffsets[1].y           = int32_t(texture.height >> i);
+image_blit.dstOffsets[1].z           = 1;
+```
+
+Before we can blit to this mip level, we need to transition it's image layout to ```TRANSFER_DST```:
+```cpp
+// Prepare current mip level as image blit destination
+vkb::insert_image_memory_barrier(
+    blit_command,
+    texture.image,
+    0,
+    VK_ACCESS_TRANSFER_WRITE_BIT,
+    VK_IMAGE_LAYOUT_UNDEFINED,
+    VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,
+    VK_PIPELINE_STAGE_TRANSFER_BIT,
+    VK_PIPELINE_STAGE_TRANSFER_BIT,
+    {VK_IMAGE_ASPECT_COLOR_BIT, i, 1, 0, 1});
+``` 
+Note that we set the ```baseMipLevel``` of the subresource range to ```i```, so the image memory barrier will only affect the one mip level we want to copy to.
+
+Now that the mip level we want to copy from and the one we'll copy to are in the proper layout (transfer source and destination) we can issue the [```vkCmdBlitImage```](https://www.khronos.org/registry/vulkan/specs/1.0/man/html/vkCmdBlitImage.html) to copy from mip level (i-1) to mip level (i):
+
+```cpp
+vkCmdBlitImage(
+    blit_command,
+    texture.image,
+    VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
+    texture.image,
+    VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,
+    1,
+    &image_blit,
+    VK_FILTER_LINEAR);
+```
+```vkCmdBlitImage``` does the down sampling from mip level (i-1) to mip level (i) using a linear filter, if you need better or more advanced filtering for this you need to resort to compute shaders for generating the mip chain instead of blitting.
```

Comment:
Compute shaders wouldn't really be the only option to generate the mip chain in that case.
We could say "you need to resort to using custom shaders", implying that you could do it either via fragment or compute.

Reply:
I have changed the wording accordingly.